cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/SchemaService.java

```diff
@@ -0,0 +1,208 @@
+package io.aiven.klaw.clusterapi.services;
+
+import io.aiven.klaw.clusterapi.models.ApiResponse;
+import io.aiven.klaw.clusterapi.models.ClusterSchemaRequest;
+import io.aiven.klaw.clusterapi.models.ClusterStatus;
+import io.aiven.klaw.clusterapi.models.KafkaClustersType;
+import io.aiven.klaw.clusterapi.models.KafkaSupportedProtocol;
+import io.aiven.klaw.clusterapi.utils.ClusterApiUtils;
+import java.util.*;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+@Slf4j
+public class SchemaService {
+  public static final String SCHEMA_REGISTRY_CONTENT_TYPE =
+      "application/vnd.schemaregistry.v1+json";
+
+  @Value("${klaw.schemaregistry.compatibility.default:BACKWARD}")
+  private String defaultSchemaCompatibility;
+
+  final ClusterApiUtils clusterApiUtils;
+
+  public SchemaService(ClusterApiUtils clusterApiUtils) {
+    this.clusterApiUtils = clusterApiUtils;
+  }
+
+  public synchronized ApiResponse registerSchema(ClusterSchemaRequest clusterSchemaRequest) {
+    try {
+      log.info("Into register schema request {}", clusterSchemaRequest);
+      //            set default compatibility
+      //            setSchemaCompatibility(environmentVal, topicName, false, protocol);
+      String suffixUrl =
+          clusterSchemaRequest.getEnv()
+              + "/subjects/"
+              + clusterSchemaRequest.getTopicName()
+              + "-value/versions";
+      Pair<String, RestTemplate> reqDetails =
+          clusterApiUtils.getRequestDetails(
+              suffixUrl, clusterSchemaRequest.getProtocol(), KafkaClustersType.SCHEMA_REGISTRY);
+
+      Map<String, String> params = new HashMap<>();
+      params.put("schema", clusterSchemaRequest.getFullSchema());
+
+      HttpHeaders headers = new HttpHeaders();
+      headers.set("Content-Type", SCHEMA_REGISTRY_CONTENT_TYPE);
+      HttpEntity<Map<String, String>> request = new HttpEntity<>(params, headers);
+      ResponseEntity<String> responseNew =
+          reqDetails.getRight().postForEntity(reqDetails.getLeft(), request, String.class);
+
+      String updateTopicReqStatus = responseNew.getBody();
+      log.info(responseNew.getBody());
+
+      return ApiResponse.builder().result(updateTopicReqStatus).build();
+    } catch (Exception e) {
+      log.error(e.getMessage());
```

Comment:
```suggestion
      log.error("Exception:", e);
```
it's better to log the whole trace